compiler/prefs.js

```diff
@@ -1,29 +1,813 @@
+    target: 'compileAllocLog'
+  },
+  {
+    arg: 'runtime-alloc-log',
+    description: 'Print information at runtime about allocated memory',
+    target: 'compileAllocLog'
+  },
+  {
+    arg: 'exception-mode',
+    type: [ 'lut', 'stack', 'stackest', 'partial' ],
+    description: 'Change the way exceptions are represented',
+    target: 'exceptionMode'
+  },
+  {
+    arg: 'todo-time',
+    type: [ 'compile', 'runtime' ],
+    description: 'Whether to emit TODO errors at compile time or runtime',
+    target: 'todoTime'
+  },
+  {
+    arg: 'truthy',
+    type: [ 'full', 'no_negative', 'no_nan_negative' ],
+    description: 'Determines what is considered "truthy" as in when an if statement passes',
+    target: 'todoTime'
+  },
+  {
+    arg: 'code-log',
+    description: 'Print information during the code generation process',
+    target: 'codeLog'
+  },
+  {
+    arg: 'always-value-internal-throws',
+    description: 'Internal throws always return a value',
+    target: 'alwaysValueInternalThrows'
+  },
+  {
+    arg: 'scoped-page-names',
+    description: 'Make allocation page names depend on scope name',
+    target: 'scopedPageNames'
+  },
+  {
+    arg: 'typeswitch-brtable',
+    description: 'Perform type switch using the br_table Wasm instruction',
+    target: 'typeswitchBrtable'
+  },
+  {
+    arg: 'typeswitch-unique-tmp',
+    description: 'Uniquify temporary variables for typeswitches using ids',
+    target: 'typeswitchUniqueTmp'
+  },
+  {
+    arg: 'indirect-call-mode',
+    type: [ 'strict', 'vararg' ],
+    description: 'Determines how indirect calls are done',
+    target: 'indirectCallMode'
+  },
+  {
+    arg: 'indirect-call-min-argc',
+    paramName: 'count',
+    type: 'int',
+    description: 'Determines how many arguments a function will at least have for call mode vararg',
+    target: 'indirectCallMinArgc'
+  },
+  {
+    arg: 'fast-length',
+    description: 'Use a shortcut for length members, assuming that the object is valid',
+    target: 'fastLength'
+  },
+  {
+    arg: 'warn-assumed-type',
+    description: 'Print a warning if a certain type is assumed during compilation',
+    target: 'warnAssumedType'
+  },
+  {
+    arg: 'backtrace-surrounding',
+    paramName: 'count',
+    type: 'int',
+    description: 'How much Wasm code should be shown before / after the current position during a backtrace',
+    target: 'backtraceSurrounding'
+  },
+  {
+    arg: 'backtrace-func',
+    description: 'Instead of showing only a small portion of the function for backtraces, show it entirely',
+    target: 'backtraceFunc'
+  },
+  {
+    arg: 'regex-log',
+    description: 'Print debug information about Wasm generated from regular expressions',
+    target: 'regexLog'
+  },
+  {
+    short: '%',
+    description: 'Print percentages in profiling mode',
+    target: 'percent'
+  }
+];
 
-  for (const x of process.argv) {
-    if (x[0] !== '-') continue;
+export function parseArgs(argv, optionsOnly = false, file) {
```

Comment:
Usually Porffor uses the `export const parseArgs = (...) => ...` format, not `function`

Reply:
Okay, I guess a simple change.